### PR TITLE
email notification UI added

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**
Initiated email notification feature

### CODE REVIEW

File: src/domain/insert_email.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import validators
+from PySide6.QtWidgets import QDialog, QDialogButtonBox
+from ui.ui_insert_email import Ui_DialogInsertEmail
```
Comment (cdimauro) [inline src/domain/insert_email.py:3]: from PySide6.QtWidgets import QDialog, QDialogButtonBox
import validators

from ui.ui_insert_email import Ui_DialogInsertEmail

File: src/domain/insert_email.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+class DialogInsertEmail(QDialog, Ui_DialogInsertEmail):
+    def __init__(self):
+        super(DialogInsertEmail, self).__init__()
```
Comment (cdimauro) [inline src/domain/insert_email.py:7]: super().__init__()

File: src/domain/insert_email.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        """Check if inserted email config is valid."""
+
+        if self.valid_email and self.valid_smtp and self.valid_port and self.valid_password:
+            self.ui.label_status.setText("")
+            self.ui.buttonBox.button(QDialogButtonBox.Ok).setEnabled(True)
+        else:
+            self.ui.buttonBox.button(QDialogButtonBox.Ok).setEnabled(False)
+   
+    def validate_smtp_port(self):
+        if self.ui.lineEdit_port.text().isdigit() and 1 <= int(self.ui.lineEdit_port.text()) <= 65535:
```
Comment (cdimauro) [inline src/domain/insert_email.py:46]: if (text := self.ui.lineEdit_port.text()).isdigit() and 1 <= int(text) <= 65535:

File: src/domain/insert_url.py [hunk 4]
```diff
@@ -30,14 +27,3 @@ def validate_url(self):
             self.ui.buttonBox.button(QDialogButtonBox.Ok).setEnabled(True)
         else:
             self.ui.buttonBox.button(QDialogButtonBox.Ok).setEnabled(False)
```
Comment (cdimauro) [inline src/domain/insert_url.py:29]: is_validated = bool(validators.url(url))
self.ui.buttonBox.button(QDialogButtonBox.Ok).setEnabled(is_validated)

File: src/domain/mainwindow.py [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 import logging
```
Comment (cdimauro) [inline src/domain/mainwindow.py:3]: import logging
import os

import keyring
from PySide6 import QtGui
from PySide6.QtCore import QThread
from PySide6.QtWidgets import QMainWindow

from domain.insert_email import DialogInsertEmail
from domain.insert_url import InsertUrlDialog
from domain.operation_mode import OperationMode
from domain.qtextedit_logger import QTextEditLogger
from domain.select_mode import DialogSelectMode
from domain.test_model_mode import TestModelMode
from ui.ui_mainwindow import Ui_MainWindow

File: src/domain/mainwindow.py [hunk 6]
```diff
@@ -31,11 +33,12 @@ def __init__(self):
         self.operation_mode_name = ""
         self.ai_model = None
         self.operation_mode = None
+        self.key_ring = None
+        self.email = {}
 
         # Connect Actions
```
Comment (cdimauro) [inline src/domain/mainwindow.py:39]: Redundant comment.

File: src/requirements.txt [hunk 7]
```diff
@@ -1,3 +1,4 @@
+keyring
 ipykernel
```
Comment (cdimauro) [inline src/requirements.txt:2]: ipykernel
keyring

File: src/requirements.txt [hunk 8]
```diff
@@ -1,5 +1,5 @@
-keyring
 ipykernel
+keyring
 timm
 validators
 torch
```
Comment (cdimauro) [inline src/requirements.txt:5]: Move to L4 (before validators)

File: src/ui/ui_insert_email.py [hunk 9]
```diff
@@ -75,7 +75,7 @@ def setupUi(self, DialogInsertEmail):
 
         self.lineEdit_password = QLineEdit(self.gridLayoutWidget)
         self.lineEdit_password.setObjectName(u"lineEdit_password")
```
Comment (cdimauro) [inline src/ui/ui_insert_email.py:77]: Why using u""? Is this automatically generated code?

File: src/ui/ui_insert_email.py [hunk 10]
```diff
@@ -92,6 +92,10 @@ def setupUi(self, DialogInsertEmail):
         self.label_status = QLabel(DialogInsertEmail)
         self.label_status.setObjectName(u"label_status")
```
Comment (cdimauro) [inline src/ui/ui_insert_email.py:93]: Same here: u"".

File: src/ui/ui_insert_email.py (hunk 9 above)
Comment (stefanodellosa-personal) [reply src/ui/ui_insert_email.py:77]: ui_ python files under ui folder are autogenerated by qtdesigner. I don't touche them as they get overwritten anytime i change UI. we can skip such files from review..

File: src/ui/ui_insert_email.py (hunk 10 above)
Comment (stefanodellosa-personal) [reply src/ui/ui_insert_email.py:93]: autogenerated file

File: src/domain/insert_email.py [hunk 11]
```diff
@@ -1,76 +1,128 @@
+"""Email configuration module."""
+
 from PySide6.QtWidgets import QDialog, QDialogButtonBox
+import keyring
```
Comment (cdimauro) [inline src/domain/insert_email.py:4]: import keyring
from PySide6.QtWidgets import QDialog, QDialogButtonBox

**Pull request merged**